server/lib/tuist/authorization.ex

```diff
@@ -174,6 +175,12 @@ defmodule Tuist.Authorization do
     action :update do
       desc("Allows an account token with project_qa_run_update scope to update a QA run.")
       allow([:authenticated_as_account, scopes_permit: :project_qa_run_update])
+
+      desc(
+        "Allows the authenticated project to update the QA run if it matches the project for which the QA run belongs."
+      )
+
+      allow([:authenticated_as_project, :projects_match])
```

Comment:
why is this needed? We shouldn't receive requests that update the QA run with the project token 🤔